server/src/test/java/io/crate/execution/ddl/tables/AddColumnTaskTest.java

```diff
@@ -332,4 +332,38 @@ public void test_table_version_less_than_5_5_oid_is_not_assigned() throws Except
             assertThat(addedColumn).isReference().hasOid(COLUMN_OID_UNASSIGNED);
         }
     }
+
+    @Test
+    public void test_cannot_add_column_that_clashes_with_index() throws Exception {
```

Comment:
Wrote this as a AddColumnTaskTest test case instead of DocTableInfoTest to make it easier to backport to 5.5 where the logic is in the task